ipatests/test_integration/test_dns_locations.py

```diff
@@ -83,53 +125,83 @@ def install(cls, mh):
         # give time to named to retrieve new records
         time.sleep(20)
 
-    def _test_against_server(self, server_ip, domain, expected_servers):
-        for rname, port in IPA_DEFAULT_MASTER_SRV_REC:
-            name_abs = rname.derelativize(domain)
-            expected = _gen_expected_srv_rrset(
-                name_abs, port, expected_servers)
-            query = resolve_records_from_server(
-                name_abs, 'SRV', server_ip)
-            assert expected == query, (
-                "Expected and received DNS data do not match on server "
-                "with IP: '{}' for name '{}' (expected:\n{}\ngot:\n{})".format(
-                    server_ip, name_abs, expected, query))
+    @classmethod
+    def delete_update_records(cls, rec_list=IPA_DEFAULT_MASTER_SRV_REC):
+        filepath = '/tmp/ipa.nsupdate'
+
+        cls.master.run_command([
+            'ipa', 'dns-update-system-records', '--dry-run', '--out', filepath
+        ])
+
+        if rec_list == IPA_CA_A_REC:
```

Comment:
Please do not use such comparison. Remove that ifs completly please.

This method needs only `rname`, so ask only for rnames as parameter, not for the whole structure.

```
delete_update_records(rnames=(r[0] for r in IPA_DEFAULT_MASTER_SRV_REC))
delete_update_records(rnames=IPA_CA_A_REC)
```

This method should be probably called `delete_update_system_records` as it is updating only them.

Reply:
makes sense...